enforce-pr-title/action.yaml

```diff
@@ -0,0 +1,63 @@
+name: enforce-pr-title
+description: Enforce pull request title includes Jira Issue key
+
+inputs:
+  repository-ref:
+    description: "Repository ref"
+    required: false
+    default: master
+
+  repository-access-token:
+    description: "Repository access token"
+    required: true
+
+  jira-account-name:
+    description: "Jira login name"
+    required: true
+
+  jira-token:
+    description: "Jira access token generated by configured account above"
+    required: true
+
+runs:
+  using: composite
+  steps:
+
+    - name: Validate PR title
+      shell: bash -l -ET -eo pipefail {0}
+      env:
+        GITHUB_TOKEN: ${{ inputs.repository-access-token }}
+        GITHUB_REF: ${{ inputs.repository-ref }}
+        JIRA_USER: ${{ inputs.jira-account-name }}
+        JIRA_TOKEN: ${{ inputs.jira-token }}
+      run: |
+        RED='\033[0;31m'
+        GREEN='\033[0;32m'
+        
+        PR_NUMBER=$(echo $GITHUB_REF | awk 'BEGIN { FS = "/" } ; { print $3 }')
+        
+        PR_TITLE=$(curl -s -H "Authorization: token $GITHUB_TOKEN" \
+                    -H "Accept: application/vnd.github.v3+json" \
+                    "https://api.github.com/repos/$GITHUB_REPOSITORY/pulls/$PR_NUMBER" | jq -r '.title')
+        
+        if [[ !("$PR_TITLE" =~ ^([A-Z]){3}-{1}([0-9])*:{1}.*$) ]]; then
```

Comment:
I've comments regarding the PR title. 
At the moment there is no convention and everyone team names the title differently. 

However, there is an industry convention that we follow at FRMAL which is the [ conventional commit ](https://www.conventionalcommits.org/en/v1.0.0/) and name the title accordingly.

Since the PR title will be used for the merge commit message for squash and merge by Github we use the conventional commit message in the title. 

Our format is:
`feat(FRA-123): new feature`
or
`test(FRA-999): add new tests`

Can we enforce the convention commit message for the title so the commit messages will be unified?

Reply:
What about ticket numbers that have different patterns and above regex does not match like `INTEL-1234`?